### DISCUSSION TITLE
Creating a type that accepts a generic map

### DISCUSSION THREAD

**Discussion by bradherman** [mobxjs/mobx-state-tree#1928] [Q&A] [answered]
The API I'm working with uses Rails on the backend and serializes the errors into a generic map of `field_name: [arr_of_messages]`...  When creating an Error type, I would love to just do the following:

```
const resp = await makeMyApiCall
if (resp.type === 'errors') {
  return Error.create(resp.data)
}
return OtherModel.create(resp.data
```

If the payload looks like this:

```
{
  "data" => {
    "type" => "errors",
    "attributes" => {
        "errors" => {
          "terms_of_service" => ["can't be blank"],
          "email" => ["too short", "must contain @"],
          ...
```
how can I define my Error model to accept the errors attribute directly knowing the keys in the map will be dynamic?

```
export const Error = types.model('Error', {
  errors: types.map(...),
})
```

**Comment by EmilTholin** [comment, marked as answer]
Hi @bradherman!

You could model it as a map of array of strings:

```js
export const Error = types.model('Error', {
  errors: types.map(types.array(types.string)),
})
```

**Reply by bradherman** [reply]
will give that a shot... thanks!
